### PR TITLE
Add HTML.Noopener to add a noopener rel to every external link

### PR DESCRIPTION
**Pull request by bashofmann**
This has performance benefits https://jakearchibald.com/2016/performance-benefits-of-rel-noopener/ but most importantly also security benefits https://mathiasbynens.github.io/rel-noopener/

Adresses https://github.com/ezyang/htmlpurifier/issues/96

### CODE REVIEW

**Comment by ezyang** [comment]
@bashofmann I think a strong case could be made for turning this on by default with %HTML.TargetBlank (but keeping this configuration option to let users turn it off to get the old behavior.) Can we do something like that?

**Comment by bashofmann** [comment]
@ezyang Sure, if you prefer it. I'll try to get to it later today or over the weekend.

**Comment by bashofmann** [comment]
I implemented it now similarly to HTML.TargetNoreferrer

File: library/HTMLPurifier/AttrTransform/TargetNoopener.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+     * @return array
+     */
+    public function transform($attr, $config, $context)
+    {
+        if (isset($attr['rel'])) {
+            $rels = explode(' ', $attr['rel']);
+        } else {
+            $rels = array();
+        }
+        if (isset($attr['target']) && !in_array('noopener', $rels)) {
```
Comment (ezyang) [inline library/HTMLPurifier/AttrTransform/TargetNoopener.php]: Doesn't this unconditionally add noopener to ALL links, whether or not they have target? (test?)

Comment (bashofmann) [inline library/HTMLPurifier/AttrTransform/TargetNoopener.php]: No, it adds the noopener only if there is a target attribute. This is also already implemented the same way in TargetNoreferrer.

See also the test case here https://github.com/ezyang/htmlpurifier/pull/112/files#diff-80cece2592ea9e4b8a72189e9207d866R40

**Comment by bashofmann** [comment]
Is there anything preventing this from being merged?

**Comment by ezyang** [comment]
Nope, I was just lagging. Thanks!

**Pull request merged**